compiler/noirc_frontend/src/elaborator/variable.rs

```diff
@@ -58,16 +58,7 @@ impl Elaborator<'_> {
         let definition_kind = definition.as_ref().map(|definition| definition.kind.clone());
 
         let mut bindings = TypeBindings::default();
-
-        // Resolve any generics if we the variable we have resolved is a function
-        // and if the turbofish operator was used.
         let generics = if let Some(DefinitionKind::Function(func_id)) = &definition_kind {
-            self.resolve_function_turbofish_generics(func_id, resolved_turbofish, location)
-        } else {
-            None
-        };
-
-        if let Some(DefinitionKind::Function(func_id)) = &definition_kind {
```

Comment:
I decided to collapse this if statement with the one above it as they are checking for the same thing.